tools/release.rb

```diff
@@ -23,96 +23,68 @@
 require_relative "spec_template"
 
 class Release
-  def self.version
-    if File.exist?("version.go")
-      File.read("version.go")[/(\d+\.\d+\.\d+)/]
-    else
-      Machinery::VERSION
-    end
-  end
+  include ReleaseChecks
+
+  RPM_CHANGES_FILE = File.join(Machinery::ROOT, "package/machinery.changes")
+  CHANGELOG_FILES  = [
+    File.join(Machinery::ROOT, "NEWS"),
+    File.join(Machinery::ROOT, "RPM_CHANGES")
+  ]
 
   def initialize(opts = {})
-    @current_version = Release.version
     @options = {
-      version:      Release.generate_development_version,
-      skip_rpm_cleanup: false,
-      jenkins_name:     "machinery-unit",
-      changes_file:     "NEWS",
-      rpm_changes_file: "RPM_CHANGES",
-      package_name:     "machinery"
+      version:      generate_development_version,
+      skip_rpm_cleanup: false
     }.merge(opts)
-    @tag             = "v#{@options[:version]}"
+    @release_version = @options[:version]
+    @tag             = "v#{@release_version}"
     @release_time    = Time.now.strftime('%a %b %d %H:%M:%S %Z %Y')
     @mail            = Cheetah.run(["git", "config", "user.email"], :stdout => :capture).chomp
     @gemspec         = Gem::Specification.load("machinery.gemspec")
-    @release_checks  = ReleaseChecks.new(@tag, @options[:jenkins_name])
-    @changelog_files = [@options[:changes_file], @options[:rpm_changes_file]]
-    @package_changes_path = "package/#{@options[:package_name]}.changes"
-    set_version
-    add_default_rpm_changes_entry
-  end
-
-  def check
-    @release_checks.check
   end
 
   def prepare
     clean_up_tmp
     remove_old_releases(skip_rpm_cleanup: @options[:skip_rpm_cleanup])
-    build_man_page
-    build_gem
+    set_version
     generate_specfile
     copy_rpmlintrc
+    add_default_rpm_changes_entry
     generate_changelog
   end
 
-  def build_gem
-    if File.exist?("machinery.gemspec")
-      Cheetah.run "gem", "build", "machinery.gemspec"
-      FileUtils.mv Dir.glob("machinery-*.gem"), "package/"
-    end
-  end
-
-  def build_man_page
-    if Dir.exist?("man")
-      Cheetah.run "rake", "man_pages:build"
-    end
-  end
-
   # Update the version file, generate a changelog and RPM specfile and build
   # an RPM using osc
   def build_local
     prepare
 
-    Rake::Task["osc:build"].invoke("--clean")
+    Rake::Task['osc:build'].invoke
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.